File: backend/app/models/company.py
"""Company entity — the parent of one or more business plan projects."""
from __future__ import annotations

from pydantic import Field

from .base import TimestampedModel

# status: active | inactive | demo


class Company(TimestampedModel):
    """A company/legal entity. Projects reference it via ``company_id``."""

    company_name: str = Field(..., min_length=1, max_length=200)
    trading_name: str | None = Field(default=None, max_length=200)
    legal_name: str | None = Field(default=None, max_length=200)
    industry_sector: str | None = Field(default=None, max_length=160)
    business_description: str | None = Field(default=None, max_length=4000)
    country: str | None = Field(default=None, max_length=120)
    city: str | None = Field(default=None, max_length=120)
    website: str | None = Field(default=None, max_length=300)
    logo_path: str | None = None
    status: str = "active"
    notes: str | None = Field(default=None, max_length=2000)


class CompanySummary(TimestampedModel):
    """Company list item with aggregated, clearly-labelled project totals."""

    company_name: str
    industry_sector: str | None = None
    country: str | None = None
    city: str | None = None
    business_description: str | None = None
    status: str = "active"
    total_projects: int = 0
    active_projects: int = 0
    draft_projects: int = 0
    profile_completion_percent: int = 0

File: frontend/src/types/company.ts
/* Company entity — the parent of one or more projects. */

export type CompanyStatus = 'active' | 'inactive' | 'demo'

export interface Company {
  id: string
  company_name: string
  trading_name?: string | null
  legal_name?: string | null
  industry_sector?: string | null
  business_description?: string | null
  country?: string | null
  city?: string | null
  website?: string | null
  logo_path?: string | null
  status: CompanyStatus
  notes?: string | null
  created_at: string
  updated_at: string
}

export interface CompanySummary {
  id: string
  company_name: string
  industry_sector?: string | null
  country?: string | null
  city?: string | null
  business_description?: string | null
  status: CompanyStatus
  total_projects: number
  active_projects: number
  draft_projects: number
  profile_completion_percent: number
  created_at: string
  updated_at: string
}

export interface CompanyUpdate {
  company_name?: string
  trading_name?: string | null
  legal_name?: string | null
  industry_sector?: string | null
  business_description?: string | null
  country?: string | null
  city?: string | null
  website?: string | null
  status?: CompanyStatus
  notes?: string | null
}
